### ISSUE TITLE
Incremental training of embeddings model

### ISSUE THREAD

**Issue by thejaswiraya** [UKPLab/sentence-transformers#1836]
I have been finetuning the paraphrase-multilingual-MiniLM-L12-v2 model on a custom dataset of 500K sentence pairs using a script very similar to [this](https://github.com/UKPLab/sentence-transformers/blob/a36e6f126cf0e333e1b00eb7cfcef2863f8919ad/examples/training/quora_duplicate_questions/training_MultipleNegativesRankingLoss.py). After fine-tuning the model, I am happy with the performance of the model. 

More often than not, I get additional training data that will improve the model's performance. **How do I incrementally train the model from the last saved checkpoint on this additional training data?** I have tried re-using the same training script as the one I use for the 500K sentence pairs, but fine-tuning on only this additional training data, but the model performs very poorly. I have tried adding the additional training data to the 500K sentence pairs and re-training the model on the entire dataset and that works - model performance is better, but it is computationally expensive to retrain/finetune the model on the entire dataset for every additional training data I collect. 

Do I need to set scheduler, warmup_steps, optimizer_class, optimizer_params parameters in the fit method? Please advise.

**Comment by thejaswiraya** [comment]
@nreimers : Any ideas on how to proceed?

**Comment by aflip** [comment]
If the dataset size for the incremental training is small, most models will over-fit causing a degradation of results.  if your task is information retrieval or search, using the model base model to create an embedding, then only incrementally adding embeddings will work the best. NGT/FAISS will do the retrieval very well as long as the embedding is good. And incremental training is generally not needed unless there's some very niche stuff that is unlikely to have been part of training.
